src/OrleansProviders/Streams/Common/CommonStreamProviderSettings.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
+    /// Common stream provider settings shared by EventHubStreamProvider, MemoryStreamProvider and GeneratorStreamProvider
+    /// </summary>
+    public class CommonStreamProviderSettings
+    {
+        /// <summary>
+        /// DataMinTimeInCache setting name.
+        /// </summary>
+        public const string DataMinTimeInCacheName = "DataMinTimeInCache";
+        /// <summary>
+        /// Drfault DataMinTimeInCache
+        /// </summary>
+        public static readonly TimeSpan DefaultDataMinTimeInCache = TimeSpan.FromMinutes(5);
+        /// <summary>
+        /// Minimum time message will stay in cache before it is available for time based purge.
+        /// </summary>
+        public TimeSpan DataMinTimeInCache = DefaultDataMinTimeInCache;
+
+        /// <summary>
+        /// DataMaxAgeInCache setting name.
+        /// </summary>
+        public const string DataMaxAgeInCacheName = "DataMaxAgeInCache";
+        /// <summary>
+        /// Default DataMaxAgeInCache
+        /// </summary>
+        public static readonly TimeSpan DefaultDataMaxAgeInCache = TimeSpan.FromMinutes(30);
+        /// <summary>
+        /// Difference in time between the newest and oldest messages in the cache.  Any messages older than this will be purged from the cache.
+        /// </summary>
+        public TimeSpan DataMaxAgeInCache = DefaultDataMaxAgeInCache;
```

Comment:
what benefits do we get there? we write more code to achieve the same functionality. And every time I have to spend couple more seconds to read the properties code to make sure it is doing what I think is it doing...